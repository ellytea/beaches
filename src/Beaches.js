import React, { Component } from 'react';
import BeachCard from './BeachCard.js'

class Beaches extends Component {
  constructor() {
    super();
    this.state = {
      beaches: [],
      beachCounties: [],
      cardVisible: false
    }
  }

  componentDidMount(){
    fetch('https://whateverly-datasets.herokuapp.com/api/v1/beaches')
      .then(response => response.json())
      .then(data => {
        this.setState({
          beaches: data.beaches,
        })
      })
      .catch(error => console.log('error'))

    fetch('https://whateverly-datasets.herokuapp.com/api/v1/beachCounties')
      .then(response => response.json())
      .then(data => {
        this.setState({
          beachCounties: data.beachCounties,
        })
      })
      .catch(error => console.log('error'))
  }

  beachVisibility = () => {
      this.setState({
        cardVisible: !this.state.cardVisible
      })
  }

  render () {
    return (
      <div className='beaches-page'>
        <ul>
        {
          this.state.beaches.map((beach) => {
            return (
              <div className='beach-grid'>
                <div className='beach-card'>
                  <a onClick={this.beachVisibility} href='#'>{beach.name}</a>
                  <img className='beach-pics' src={beach.image}/>
                  {this.state.cardVisible && <BeachCard beach={beach} beachVisibility={this.beachVisibility} />}
                </div>
              </div>
            )
          })
        }
        </ul>
      </div>
    )
  }
}


export default Beaches;