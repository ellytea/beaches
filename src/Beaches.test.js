import React from 'react';
import Beaches from './Beaches.js';
import { mount, shallow } from 'enzyme';


describe('Beaches', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Beaches key= 'Beaches'/>
      )
})

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should change beachCard state to empty string when clicked', () => {
      wrapper.find('.beach-body').simulate('click');
      expect(wrapper.state('beachCard')).toEqual('');
  })

})

