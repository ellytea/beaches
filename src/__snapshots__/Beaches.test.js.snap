// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beaches should match the snapshot with all data passed in 1`] = `
<div
  className="beaches-page"
>
  <header
    className="header"
  >
    <h1
      className="header-title"
    >
      Beachy Keen
    </h1>
  </header>
  <Filter
    beachCounties={Array []}
    filterByCounty={[Function]}
  />
  <div
    className="beach-body"
  />
  <BeachesCard
    closeBeach={[Function]}
    selectedBeach=""
  />
</div>
`;
